libgstc/pygstc/gstc.py

```diff
@@ -0,0 +1,691 @@
+        Sets a property in an element of a given pipeline
+    event_eos(pipe_name)
+        Send an end-of-stream event
+    event_flush_start(pipe_name)
+        Put the pipeline in flushing mode
+    event_flush_stop(pipe_name, reset='true')
+        Take the pipeline out from flushing mode
+    event_seek (pipe_name, rate='1.0', format='3', flags='1', start_type='1', start='0', end_type='1', end='-1')
+        Perform a seek in the given pipeline
+    list_elements(pipe_name)
+        List the elements in a given pipeline
+    list_pipelines( )
+        List the existing pipelines
+    list_properties(pipe_name, element)
+        List the properties of an element in a given pipeline
+    list_signals(pipe_name, element)
+        List the signals of an element in a given pipeline
+    pipeline_create(pipe_name,  pipe_desc)
+        Creates a new pipeline based on the name and description
+    pipeline_delete(pipe_name)
+        Deletes the pipeline with the given name
+    pipeline_pause(pipe_name)
+        Sets the pipeline to paused
+    pipeline_play(pipe_name)
+        Sets the pipeline to playing
+    pipeline_stop(pipe_name)
+        Sets the pipeline to null
+    read(uri)
+        Deletes the resource held at the given URI with the given name
+    signal_connect(pipe_name, element, signal)
+        Connect to signal and wait
+    signal_disconnect(pipe_name, element, signal)
+        Disconnect from signal
+    signal_timeout(pipe_name, element, signal, timeout)
+        Apply a timeout for the signal waiting. -1: forever, 0: return
+        immediately, n: wait n microseconds
+    update(uri, value)
+        Updates the resource at the given URI
+    """
+    def __init__(self, ip='localhost', port=5000, logger=None, timeout=0):
+        """
+        Initialize new GstdClient
+
+        Parameters
+        ----------
+        ip : string
+            IP where GSTD is running
+        port : string
+            Port where GSTD is running
+        logger : CustomLogger
+            Custom logger where all log messages from this class are going
+            to be reported
+        """
+        if logger:
+            self._logger = logger
+        else:
+            self._logger = DummyLogger()
+        self._ip = ip
+        self._port = port
+        self._logger.info('Starting GStreamer Daemon Client with ip=%s port=%d'
+                           % (self._ip, self._port))
+        self._test_gstd()
+        self._ipc = Ipc(self._logger, self._ip, self._port)
+        self._timeout = timeout
+
+    def _send_cmd_line(self, cmd_line):
+        """
+        Send a command using an abstract IPC and wait for the response
+
+        Parameters
+        ----------
+        cmd_line : string list
+            Command to be send
+
+        Raises
+        ------
+        GstdError
+            Error is triggered when Gstd IPC fails
+        GstcError
+            Error is triggered when the Gstd python client fails internally
```

Comment:
Since this is the same error do we need it to be declared twice? Raises in the docstring is usually declared so that the caller function can use a particular `Except GstdError` clause.

Reply:
`GstcError != GstdError`